src/runtime/float.cpp

```diff
@@ -124,7 +124,11 @@ extern "C" Box* floatAdd(BoxedFloat* lhs, Box* rhs) {
     } else if (rhs->cls == float_cls) {
         return floatAddFloat(lhs, static_cast<BoxedFloat*>(rhs));
     } else if (rhs->cls == long_cls) {
-        return boxFloat(lhs->d + PyLong_AsDouble(rhs));
+        double rhs_f = PyLong_AsDouble(rhs);
+        if (PyErr_Occurred()) {
```

Comment:
Minor, but I think PyLong_AsDouble is supposed to return -1.0 when an exception is thrown, and checking the return value against -1.0 should be slightly faster than going to PyErr_Occurred every time.  (we still need to call it if the value is -1.0, but that should be relatively rare)